### ISSUE TITLE
Error creating tmp directory while executing perfkit benchmark from windows box.

### ISSUE THREAD

**Issue by kuldeepJS** [GoogleCloudPlatform/PerfKitBenchmarker#188]
I am trying to run google's perfkit performance tool on GCP as described on https://github.com/GoogleCloudPlatform/PerfKitBenchmarker. After getting the GCP account and project id, when I am trying to run the perfkit from my personal machine, I encountered following error:

Generating temporary directory at  /tmp/perfkitbenchmarker\run_4c0ee849
Traceback (most recent call last):
  File "pkb.py", line 21, in <module>
    sys.exit(Main())
  File "C:\Users\sony\Desktop\PerfKitBenchmarker-0.11.1\perfkitbenchmarker\pkb.py", line 358, in Main
    return RunBenchmarks()
  File "C:\Users\sony\Desktop\PerfKitBenchmarker-0.11.1\perfkitbenchmarker\pkb.py", line 291, in RunBenchmarks
    ConfigureLogging()
  File "C:\Users\sony\Desktop\PerfKitBenchmarker-0.11.1\perfkitbenchmarker\pkb.py", line 167, in ConfigureLogging
    vm_util.GenTempDir()
  File "C:\Users\sony\Desktop\PerfKitBenchmarker-0.11.1\perfkitbenchmarker\vm_util.py", line 88, in GenTempDir
    create_process = subprocess.Popen(create_cmd)
  File "C:\Python27\lib\subprocess.py", line 672, in __init__
    errread, errwrite)
  File "C:\Python27\lib\subprocess.py", line 882, in _execute_child
    startupinfo)
WindowsError: [Error 2] The system cannot find the file specified
There seems to be issue with creating a temporary directory, but should I go and create this directory? How should I run this tool from my personal machine? Personal machine configuration on which I am trying to run the perfkit: OS: Windows7 NOTE: I am running the commands in administrator mode.

**Comment by voellm** [comment]
Kuldeep - We have not run this much on Windows boxes so you are in a bit of
uncharted territory.  Are you using CYGWIN?  If not I recommend starting
there because PerfKit also uses bash, SSH, etc.

One workaround is to boot a tiny LINUX VM like an f1-micro on GCE or boot a
local Linux VM on your dev box if you have visualization software and run
from there.




On Mon, Mar 23, 2015 at 9:25 AM, Kuldeep <notifications@github.com> wrote:

> I am trying to run google's perfkit performance tool on GCP as described
> on https://github.com/GoogleCloudPlatform/PerfKitBenchmarker. After
> getting the GCP account and project id, when I am trying to run the perfkit
> from my personal machine, I encountered following error:
>
> Generating temporary directory at /tmp/perfkitbenchmarker\run_4c0ee849
> Traceback (most recent call last):
> File "pkb.py", line 21, in
> sys.exit(Main())
> File
> "C:\Users\sony\Desktop\PerfKitBenchmarker-0.11.1\perfkitbenchmarker\pkb.py",
> line 358, in Main
> return RunBenchmarks()
> File
> "C:\Users\sony\Desktop\PerfKitBenchmarker-0.11.1\perfkitbenchmarker\pkb.py",
> line 291, in RunBenchmarks
> ConfigureLogging()
> File
> "C:\Users\sony\Desktop\PerfKitBenchmarker-0.11.1\perfkitbenchmarker\pkb.py",
> line 167, in ConfigureLogging
> vm_util.GenTempDir()
> File
> "C:\Users\sony\Desktop\PerfKitBenchmarker-0.11.1\perfkitbenchmarker\vm_util.py",
> line 88, in GenTempDir
> create_process = subprocess.Popen(create_cmd)
> File "C:\Python27\lib\subprocess.py", line 672, in *init*
> errread, errwrite)
> File "C:\Python27\lib\subprocess.py", line 882, in _execute_child
> startupinfo)
> WindowsError: [Error 2] The system cannot find the file specified
> There seems to be issue with creating a temporary directory, but should I
> go and create this directory? How should I run this tool from my personal
> machine? Personal machine configuration on which I am trying to run the
> perfkit: OS: Windows7 NOTE: I am running the commands in administrator mode.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/issues/188>.
>



-- 
Anthony F. Voellm (aka Tony)
Google Voice:  (650) 516-7382
<https://www.google.com/voice/b/0?pli=1#phones>
Blog: http://perfguy.blogspot.com/

**Comment by kuldeepJS** [comment]
I tried it out and the test is running but they all are failing with the message:
2015-03-24 04:37:17,738 INFO     Running: gcloud compute disks describe perfkit-7a8dab8a-0 --project poc-cld-test --format json --quiet --zone us-central1-a
2015-03-24 04:37:18,087 ERROR    Got exception running _CreateResource: Creation of GceDisk failed.
2015-03-24 04:37:38,573 INFO     Running: gcloud compute disks create perfkit-7a8dab8a-0 --size 10 --type pd-standard --project poc-cld-test --format json --quiet --zone us-central1-a --image ubuntu-14-04

**Comment by voellm** [comment]
Does:

gcloud compute --project poc-cld-test instances list

work?  It looks like gcloud is not working.

**Comment by kuldeepJS** [comment]
I got through it. The reason for such a behaviour was:
1. Executing gcloud gauth login; creates a .boto file
2. Executing the second step for GCE: $ gsutil config; called gauth and reset the authentication key in .boto file but the gcloud environment is still logged in using the authentication created in step 1.

Thus I ran step 1 again and it started working.

I was able to execute ping benchmark successfully.
